lib/AST/ASTPrinter.cpp

```diff
@@ -2610,7 +2636,25 @@ void PrintAST::printOneParameter(const ParamDecl *param, bool Curried,
   // Set and restore in-parameter-position printing of types
   {
     llvm::SaveAndRestore<bool> savePrintParam(Options.PrintAsInParamType, true);
+    // FIXME: don't do if will be using type repr printing
+    printParameterFlags(Printer, Options, paramFlags);
+
+    // Special case, if we're not going to use the type repr printing, peek
+    // through the paren types so that we don't print excessive @escapings
```

Comment:
Ah, I didn't realize there was a convention there
(period in this comment intentionally absent just to drive you crazy :-)